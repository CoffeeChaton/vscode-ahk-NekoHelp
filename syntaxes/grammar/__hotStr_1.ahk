; not flag
::ahk,,::AutoHotkey
::AHK,,::AutoHotkey
::dx9,,::DirectX
::ts,,::TypeScript
::es,,::eslint
::pr,,::prettier ; dddddddddd
;                ^^^^^^^^^^^^ comment


; X-flag
:x:dd,,::MsgBox, % "dd" ;


; with out r/t flag
:*b0:<em1>::</em>{left 5} {Enter} ^t ;
:*b0:<em2>::</em>{left 5} {}} </em5> {{}
:*b0:a1,::{{} ; send "{"
;            ^comment
:*b0:a2,::{}} ; send "}"
::a99,,:: </em>{left 5} {Enter} ^t ;


; r/t flag
:r:<em3>::</em>{left 5}
;              ^^^^^^^ just string

:t:<em4>::</em>{left 5}
;              ^^^^^^^ just string



; r/t + x ---> only x flag
:xr:dd5,,::MsgBox, % "dd" ; show MsgBox
:xr:dd6,,::MsgBox, % "{left 5}" ; show MsgBox
