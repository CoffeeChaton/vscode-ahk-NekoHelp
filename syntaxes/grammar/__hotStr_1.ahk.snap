>; not flag
#^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>::ahk,,::AutoHotkey
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#       ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#         ^^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
>::AHK,,::AutoHotkey
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#       ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#         ^^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
>::dx9,,::DirectX
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#       ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#         ^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
>::ts,,::TypeScript
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#      ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#        ^^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
>::es,,::eslint
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#      ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#        ^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
>::pr,,::prettier ; dddddddddd
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#      ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#        ^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                 ^^^^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk comment.line.hotstring.ahk
>;                ^^^^^^^^^^^^ comment
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>
>; X-flag
#^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>:x:dd,,::MsgBox, % "dd" ;
#^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
# ^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk entity.name.function.hotstring.x.flag.ahk
#  ^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
#   ^^^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk string.entity.name.function.hotstring.trigger.x.flag.ahk.ahk
#       ^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
#         ^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#               ^ source.ahk meta.command.line.ahk punctuation.ahk
#                ^ source.ahk meta.command.line.ahk
#                 ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#                  ^ source.ahk meta.command.line.ahk
#                   ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#                    ^^ source.ahk meta.command.line.ahk string.quoted.double.ahk
#                      ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                       ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#                        ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>
>; with out r/t flag
#^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>:*b0:<em1>::</em>{left 5} {Enter} ^t ;
#^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
# ^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk entity.name.function.hotstring.with_rt.flag.ahk
#    ^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#     ^^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#          ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#            ^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                 ^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                         ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                          ^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                                 ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                  ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                                   ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                    ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                     ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk comment.line.hotstring.ahk
>:*b0:<em2>::</em>{left 5} {}} </em5> {{}
#^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
# ^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk entity.name.function.hotstring.with_rt.flag.ahk
#    ^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#     ^^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#          ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#            ^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                 ^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                         ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                          ^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                             ^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                     ^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
>:*b0:a1,::{{} ; send "{"
#^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
# ^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk entity.name.function.hotstring.with_rt.flag.ahk
#    ^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#     ^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#        ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#          ^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#             ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#              ^^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk comment.line.hotstring.ahk
>;            ^comment
#^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>:*b0:a2,::{}} ; send "}"
#^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
# ^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk entity.name.function.hotstring.with_rt.flag.ahk
#    ^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#     ^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#        ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#          ^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#             ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#              ^^^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk comment.line.hotstring.ahk
>::a99,,:: </em>{left 5} {Enter} ^t ;
#^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#  ^^^^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk string.entity.name.function.hotstring.trigger.with_rt.flag.ahk
#       ^^ source.ahk hotstring.normal.flag.ahk keyword.operator.hotstring.with_rt.flag.ahk
#         ^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#               ^^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                       ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                        ^^^^^^^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                               ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk entity.name.function.hotstring.normal.flag.ahk
#                                 ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                  ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk
#                                   ^ source.ahk hotstring.normal.flag.ahk string.hotstring.with_rt.flag.ahk comment.line.hotstring.ahk
>
>
>; r/t flag
#^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>:r:<em3>::</em>{left 5}
#^ source.ahk hotstring.r.flag.ahk keyword.operator.hotstring.r.flag.ahk
# ^ source.ahk hotstring.r.flag.ahk keyword.operator.hotstring.r.flag.ahk entity.name.function.hotstring.r.flag.ahk
#  ^ source.ahk hotstring.r.flag.ahk keyword.operator.hotstring.r.flag.ahk
#   ^^^^^ source.ahk hotstring.r.flag.ahk keyword.operator.hotstring.r.flag.ahk string.entity.name.function.hotstring.trigger.r.flag.ahk
#        ^^ source.ahk hotstring.r.flag.ahk keyword.operator.hotstring.r.flag.ahk
#          ^^^^^^^^^^^^^ source.ahk hotstring.r.flag.ahk string.hotstring.r.flag.ahk
>;              ^^^^^^^ just string
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>:t:<em4>::</em>{left 5}
#^ source.ahk hotstring.t.flag.ahk keyword.operator.hotstring.t.flag.ahk
# ^ source.ahk hotstring.t.flag.ahk keyword.operator.hotstring.t.flag.ahk entity.name.function.hotstring.t.flag.ahk
#  ^ source.ahk hotstring.t.flag.ahk keyword.operator.hotstring.t.flag.ahk
#   ^^^^^ source.ahk hotstring.t.flag.ahk keyword.operator.hotstring.t.flag.ahk string.entity.name.function.hotstring.trigger.t.flag.ahk
#        ^^ source.ahk hotstring.t.flag.ahk keyword.operator.hotstring.t.flag.ahk
#          ^^^^^^^^^^^^^ source.ahk hotstring.t.flag.ahk string.hotstring.t.flag.ahk
>;              ^^^^^^^ just string
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>
>
>; r/t + x ---> only x flag
#^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>:xr:dd5,,::MsgBox, % "dd" ; show MsgBox
#^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
# ^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk entity.name.function.hotstring.x.flag.ahk
#   ^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
#    ^^^^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk string.entity.name.function.hotstring.trigger.x.flag.ahk.ahk
#         ^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
#           ^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#                 ^ source.ahk meta.command.line.ahk punctuation.ahk
#                  ^ source.ahk meta.command.line.ahk
#                   ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#                    ^ source.ahk meta.command.line.ahk
#                     ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#                      ^^ source.ahk meta.command.line.ahk string.quoted.double.ahk
#                        ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                         ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#                          ^^^^^^^^^^^^^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>:xr:dd6,,::MsgBox, % "{left 5}" ; show MsgBox
#^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
# ^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk entity.name.function.hotstring.x.flag.ahk
#   ^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
#    ^^^^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk string.entity.name.function.hotstring.trigger.x.flag.ahk.ahk
#         ^^ source.ahk hotstring.x.flag.ahk keyword.operator.hotstring.x.flag.ahk
#           ^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#                 ^ source.ahk meta.command.line.ahk punctuation.ahk
#                  ^ source.ahk meta.command.line.ahk
#                   ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#                    ^ source.ahk meta.command.line.ahk
#                     ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#                      ^^^^^^^^ source.ahk meta.command.line.ahk string.quoted.double.ahk
#                              ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#                               ^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk
#                                ^^^^^^^^^^^^^ source.ahk meta.command.line.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>